src/webapi/html_elements/canvas.rs

```diff
@@ -0,0 +1,108 @@
+        js! (
+            return @{self}.height;
+        ).try_into().unwrap()
+    }
+
+    /// Sets a positive integer reflecting the height HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 150 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn set_height( &self, value: u32 ) {
+        js! { @(no_return)
+            @{self}.height = @{value};
+        }
+    }
+
+    /// Returns a positive integer reflecting the width HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 300 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn width( &self ) -> u32 {
+        js! (
+            return @{self}.width;
+        ).try_into().unwrap()
+    }
+
+    /// Sets a positive integer reflecting the width HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 300 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn set_width( &self, value: u32 ) {
+        js! { @(no_return)
+            @{self}.width = @{value};
+        }
+    }
+
+    /// Returns a drawing context on the canvas, or None if the context identifier is not supported.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext)
+    pub fn get_context<T: RenderingContext>( &self ) -> Option<T> {
+        js! (
+            return @{self}.getContext(@{T::CONTEXT_TYPE});
+        ).try_into().ok().and_then(|r| T::from_reference(r))
+    }
+
+    /// Returns a data URI containing a representation of the image in the format specified by the
+    /// type parameter (defaults to PNG). The returned image is in a resolution of 96 dpi.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataUrl)
+    pub fn to_data_url( &self, mime_type: Option<&str>, quality: Option<f64> ) -> String {
+        js! (
+            return @{self}.toDataUrl(@{mime_type}, @{quality});
+        ).try_into().unwrap()
+    }
+
+    /// Creates a Blob object representing the image contained in the canvas; this file may be
+    /// cached on the disk or stored in memory at the discretion of the user agent.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob)
+    pub fn to_blob<F: FnMut(Blob) + 'static>( &self, f: F, mime_type: Option<&str>, quality: Option<f64> ) {
```

Comment:
`FnOnce` (There is a wrapper called `Once` which allows passing `FnOnce` functions to `js!`)

Reply:
Also, semi related question - do you know why the original JS API takes a callback here instead of returning a `Blob`? Is it because the `Blob` is done asynchronously?

Reply:
Yes.